### ISSUE TITLE
Object-shorthand do not ignore event handlers defined as function instead of shorthand

### ISSUE THREAD

**Issue by roman4e** [eslint/eslint#10402]
**What rule do you want to change?**
object-shorthand

**Does this change cause the rule to produce more or fewer warnings?**
Should produce fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
I don't know

**Please provide some example code that this change will affect:**

```js
document.addEventListener('click', (ev) => {
   console.log(this); // Window
   console.log(this.navigator.appName); // -> "Mozilla"
});
document.addEventListener('click', function (ev) {
   console.log(this); // #document
   console.log(this.body); // --> HTMLElement <body>...</body>
   console.log(this.navigator.appName); // -> Error: Cannot read property 'appName' of undefined
// because now this is `document`
});
```

**What does the rule currently do for this code?**
The rule warns if I use `function()` instead of `() =>` to make callback which will change its context on call

**What will the rule do after it's changed?**
Ignore `function()` callback definition when it used for event handlers at least

**Comment by platinumazure** [comment]
Hi @roman4e, thanks for the issue!

I'm a bit confused. The `object-shorthand` rule does not actually have anything to do with arrow functions. Instead it enforces that you follow certain shorthand configurations in members of object expressions:

```js
const obj = {
    // Long form:
    a: function(foo) { /* etc. */ },

    // Shorthand:
    b(foo) { /* etc. */ }
};
```

The scenario you describe seems like it might be coming from `prefer-arrow-callback`.

Could you please show the full, exact error message ESLint is showing in your case?

**Comment by roman4e** [comment]
I apologize confused you. My real case was about using the function from object created as in yours example in Long form, then used as
`    let selection = chooseRandomly(['a', 'b']);`

`    document.addEventListener('click', obj[selection]);`
And at the creation point had the warning about shorthand usage.

Seems that's my mistake.
